src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -41,6 +41,11 @@ public partial class ConfigurationManager<T> : BaseConfigurationManager, IConfig
         internal TimeProvider TimeProvider = TimeProvider.System;
         internal ITelemetryClient TelemetryClient = new TelemetryClient();
 
+        /// <summary>
+        /// Gets or sets the optional configuration event handler.
+        /// </summary>
+        public IConfigurationEventHandler<T> ConfigurationEventHandler { get; set; }
```

Comment:
Should this have a public setter?

Reply:
Number of ConfigurationManager ctors is relatively high, where each throw an exception if a null argument is provided.
Removing the setter (or the public property) would require us to double the number of ctors.

I tried to strike a compromise by allowing for both ctor and property injection (in case where not every param is provided when constructing the class).